### ISSUE TITLE
DEBUG setting not honoured

### ISSUE THREAD

**Issue by djjudas21** [daledavies/jump#113]
I'm running Jump on Kubernetes and I noticed that debug mode remains enabled even when I set `DEBUG: false`. The debug bar remains in the corner. I tried different ways of quoting `false` and I even tried setting it to `true`, but I couldn't get rid of the debug bar. Pretty sure I didn't have this on v1.3.x.

This is my deployment: https://jump.gazeley.uk/ (it's a work in progress so I'm still using the default sites but I'm going to stop fiddling for tonight)

This is the config that's being passed to the container:

```
  Containers:
   jump:
    Image:      daledavies/jump:v1.4.0
    Port:       8080/TCP
    Host Port:  0/TCP
    Requests:
      cpu:      2m
      memory:   24Mi
    Liveness:   http-get http://:http/ delay=0s timeout=1s period=10s #success=1 #failure=3
    Readiness:  http-get http://:http/ delay=0s timeout=1s period=10s #success=1 #failure=3
    Environment:
      ALTLAYOUT:       false
      AMPMCLOCK:       true
      BGBLUR:          50
      BGBRIGHT:        90
      CACHEBYPASS:     false
      CHECKSTATUS:     false
      CUSTOMGREETING:  Some other greeting message
      CUSTOMWIDTH:     1000
      DEBUG:           false
      METRICTEMP:      true
      NOINDEX:         true
      SHOWCLOCK:       false
      SHOWGREETING:    false
      SHOWSEARCH:      true
      SITENAME:        Gazeley UK
      STATUSCACHE:     5
    Mounts:
      /backgrounds from backgrounds (rw)
      /favicon from favicon (rw)
      /search/searchengines.json from search-config (rw,path="searchengines.json")
      /sites/sites.json from sites-config (rw,path="sites.json")
```

https://jump.gazeley.uk/

**Comment by daledavies** [comment]
Hmm, thanks for raising the issue. This is a new feature with v1.4.0, I've not seen this happen before in testing though. 

Let me take a look tomorrow.

**Comment by daledavies** [comment]
Wondering if you could try deleting the control the cache directory?

**Comment by daledavies** [comment]
Also can you exec into the container and check if the DEBUG environment variable is definitely false? 

You shouldn't have to specify it at all as the app sets debug to false internally if the environment variable isn't set.

**Comment by djjudas21** [comment]
I'll do these actions tomorrow. If they don't fix it, I'd put money on it being the classic bug where a value of "false" appears to be true because it is a non-empty string 😄

**Comment by daledavies** [comment]
You know what it probably is! I've even written and used a method to parse booleans from the yaml... https://github.com/daledavies/jump/blob/465d9b9d6dd9aa4b3ffff010e2d839e2b8b2f9af/jumpapp/classes/Config.php#L142

But looks like I've forgotten to use it here... 

https://github.com/daledavies/jump/blob/465d9b9d6dd9aa4b3ffff010e2d839e2b8b2f9af/jumpapp/classes/Main.php#L47

**Comment by djjudas21** [comment]
I think you've found the smoking gun, but for the sake of completeness...

```
[jonathan@poseidon jump]$ kubectl exec -it jump-7d99fd5799-49jr5 -- bash
jump-7d99fd5799-49jr5:/var/www/html# echo $DEBUG
false
```

I also couldn't find a cache dir anywhere under `/var/www/html`, and haven't knowingly set or changed this anywhere.

**Comment by daledavies** [comment]
I've added the proper boolean check mentioned above, I'll create a new release later for it.

In terms of the cache, the default directory is... `/var/www/cache`. I'm assuming you're deploying the docker image from docker hub or building it using the supplied dockerfile?

**Comment by djjudas21** [comment]
Yes I'm using the stock docker image. I found and deleted the application cache. It didn't fix the problem with debug mode, but it really upset the application until I restarted the container. Loads of errors about it trying to open files from the cache that I had deleted.

**Comment by daledavies** [comment]
I've just added a few changes to the cache directory to stop this from happening again, will add to new release with the patch to fix debug.

**Comment by daledavies** [comment]
I'll also update the readme to clarify what exactly the default values are, as it really wasn't very clear on reflection.

**Comment by djjudas21** [comment]
You mean to tell me that it's an open source project with ambiguous or missing documentation?? That never happens!

Seriously though, I appreciate your effort tracking down and fixing these bugs :slightly_smiling_face:

**Comment by daledavies** [comment]
I've seen worse with a lot of closed source applications! But honestly writing good documentation is difficult and probably the least enjoyable part of maintaining a software project for other people to use.

I'm still over the moon knowing that anyone other than me uses Jump though it's truly awesome! 😁

**Comment by daledavies** [comment]
I've created release v1.4.1, pushed to docker hub, if you pull the `latest` or `v1.4.1` tags you'll get it.

**Comment by djjudas21** [comment]
Can confirm this is fixed in v1.4.1 :trophy:

**Comment by daledavies** [comment]
Amazing, thank you so much for reaching out. All the best 😁
